servicetest/filesystem/test_filesystem_strace.py

```diff
@@ -0,0 +1,191 @@
+
+
+# This function is used by the test framework to know where test specific
+# files should be stored
+def output_dir():
+    return TESTOUTPUT_DIR
+
+
+# This function is used by the 'agent' fixture to know where the log should
+# be stored
+def logfile_path():
+    return CURRENT_DIR + "/test-strace.log"
+
+def agent_exec_prefix():
+    return "strace"
+
+DATA = {
+    Container.CONFIG: '[{"enableWriteBuffer": false}]',
+    Container.BIND_MOUNT_DIR: "/gateways/app",
+    Container.HOST_PATH: CURRENT_DIR,
+    Container.READONLY: False
+}
+
+def get_logblock(file, startline, stopline):
+    """ This gets a block of text/logs between the startline and stopline
+    from a file and returns the text as a string to the caller. Does not
+    include startline or stopline in the actual textblock passed to the caller.
+
+    :param file: The file to read
+    :param startline: The first line to look for
+    :param stopline: The last line to look for
+    :return: Returns a textbuffer
+    """
+    with open(file, "r") as input:
+        buf = ""
+        for line in input:
+            if line.strip() == startline:
+                break
+
+        for line in input:
+            if line.strip() == stopline:
+                break
+            buf += line
+
+        return buf
+
+
+def get_strace_created_files(buf):
+    """ The function receives a logblock containing strace log lines and finds
+    the possibly created files as well as the created directories. Note that
+    open("...", O_CREAT...) also finds already existing files that are just
+    opened for write.
+
+    :param buf: The string buffer to search in
+    :return: A list of files and directories likely created
+    """
+    files = list()
+
+    for line in StringIO.StringIO(buf):
+        print line
+        # Find all open("...", O_CREAT...) instances.
+        if re.search('^open(.*)O_CREAT', line):
+            p = re.compile('^open\(\"(.*)\",')
+            files.append(p.match(line).group(1))
+        # Find all mkdir instances.
+        if re.search('^mkdir', line):
+            p = re.compile('^mkdir\(\"(.*)\",')
+            files.append(p.match(line).group(1))
+
+    return files
+
+@pytest.mark.usefixtures("create_testoutput_dir", "dbus_launch", "agent", "assert_no_proxy")
+class TestFileSystemStrace(object):
+    """ This suite does some strace testing of the creation and deletion of
+    containers and make sure we don't create or modify files outside the areas
+    we should be working within.
+
+    The tests are performed by prefixing the softwarecontainer-agent binary
+    with strace and logging the output into a logfile and then parsing the
+    logfile for the files and directories being created. We can then check the
+    paths and make sure they are valid.
+    """
+
+    @pytest.mark.xfail("platform.release() <= \"3.18.0\"")
+    def test_bindmount_sockets_strace(self):
+        """ Test that sockets are bindmountable as expected.
+        """
```

Comment:
Not really any specific reason as far as I know, I wasn't sure if they behaved differently than other types of files so I wanted to have another test for them :)